python/cudf/cudf/core/timestamp.py

```diff
@@ -0,0 +1,389 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+import numpy as np
+import pandas as pd
+
+from cudf.core.scalar import Scalar
+
+
+class Timestamp(Scalar):
+    def __init__(
+        self,
+        *args,
+        **kwargs,
+    ):
+        pd_ts_kwargs = {k: v for k, v in kwargs.items() if k != "dtype"}
+        ts = pd.Timestamp(*args, **pd_ts_kwargs)
+        super().__init__(ts)
+
+    @property
+    def value(self) -> np.datetime64:
```

Comment:
Should the host value just be a `pandas.Timestamp`?

Reply:
So it looks like there's a `cudf.Scalar.value` which `cudf.Timestamp` inherits. It returns a `np.datetime64` object. But there's also `pd.Timestamp.value` which returns an integer.

I think your right that`cudf.Timestamp.value` should return `pd.Timestamp(super().value).value`. And I'll a separate property for access to the scalar value (np.datetime64).